### PR TITLE
Add live demo to website

### PR DESCRIPTION
**Pull request by jtbandes**
### Public-Facing Changes

Added a live demo to the mcap homepage, where you can record a mcap file in the browser and download it.

### Description

image

image

### CODE REVIEW

**Comment by defunctzombie** [comment]
This is great! Needs some styling love but is a fun way to end up with an mcap file.

**Review by ocin-rye — COMMENTED** [review]
Nice functionality! Added an issue for styling options: https://linear.app/foxglove/issue/FG-3612/design-styling-options-for-mcap-live-demo

File: website/src/components/McapRecordingDemo/McapRecordingDemo.tsx [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+    [state]
+  );
+
+  return (
+    <section className={styles.container}>
+      <div className={styles.column}>
+        <header>
+          <h2>Try it out</h2>
+          <div className={styles.subhead}>
+            Record an MCAP file right from your browser. Select sensors, create
```
Comment (esthersweon) [inline website/src/components/McapRecordingDemo/McapRecordingDemo.tsx:257]:
```suggestion
            Record xyz information in an MCAP file, and download it for a hands-on look at the MCAP file format.
```

And separate line: "Select sensor data to record in your MCAP file. All data is recorded and saved locally."

File: website/src/components/McapRecordingDemo/McapRecordingDemo.tsx [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+              <Link
+                href="#"
+                className={cx(
+                  "button",
+                  "button--success",
+                  styles.downloadButton
+                )}
+                onClick={onDownloadClick}
+              >
+                Download ({formatBytes(Number(state.bytesWritten))})
```
Comment (esthersweon) [inline website/src/components/McapRecordingDemo/McapRecordingDemo.tsx:339]:
```suggestion
                Download recording ({formatBytes(Number(state.bytesWritten))})
```

**Comment by esthersweon** [comment]
Can we visually differentiate between "keys" and "values" in the sensor output (i.e. bold - **X**: 5, **Yaw**: 38, **Messages**: 300)?

Also would be nice to provide info re: what "messages" and "chunks" mean within the context of MCAP (maybe link to GitHub repo sections?).

**Review by esthersweon — APPROVED** [review]
#nit - Visually looks a little strange that the checkbox items are a darker color than introductory text. 

Also, stand by my confusion re: the "Activate camera..." line. Maybe using "Toggle on 'Camera' ..." could be a compromise?

Copy and content look great.

File: website/src/components/McapRecordingDemo/McapRecordingDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+    <section className={styles.container}>
+      <div className={styles.column}>
+        <header>
+          <h2>Try it out</h2>
+          <p className={styles.subhead}>
+            Record sensor data to an MCAP file right now, for a hands-on look at
+            this flexible file format.
+          </p>
+          <p className={styles.subhead}>
+            Select sensor data to record in your MCAP file. All data is recorded
```
Comment (esthersweon) [inline website/src/components/McapRecordingDemo/McapRecordingDemo.tsx:261]: I didn't make this suggestion before, but seeing the 2 new lines together is making me realize that it's a bit repetitive.

```suggestion
            Select sensor data to record. All data is recorded
```

**Comment by defunctzombie** [comment]
👏 Good stuff

**Pull request merged**